### ISSUE TITLE
Autouse FileChooser based on type= argument.

### ISSUE THREAD

**Issue by anntzer** [chriskiehl/Gooey#470]
It would be nice if
```
from argparse import *
from pathlib import Path
from gooey import Gooey

@Gooey
def main():
    parser = ArgumentParser()
    parser.add_argument("--foo", type=FileType("r"))
    parser.add_argument("--bar", type=Path)
    print(parser.parse_args())

main()
```
took advantage of the `type` kwarg passed to the ArgumentParser and auto-sets the corresponding widget to a FileChooser.

**Comment by alkasm** [comment]
Had the same thought, I use `FileType` for tons of my scripts already (`Path` is also a great idea for when the path is a directory or a file that doesn't need to be opened).

**Comment by chriskiehl** [comment]
There's another of these requests floating around. Will implement in the next release!

**Comment by chriskiehl** [comment]
I went about adding this, testing it, and _almost_ committing it, before I ultimately decided against it. It currently lives on a branch here: https://github.com/chriskiehl/Gooey/tree/1.0.4-release--issue-470  

Defaulting `Filetype` to FileChooser is a _fairly_ safe default. `Path` has no clear sane default, as it's too general a thing. Automatically picking FileChooser when the user intended directory paths to be supplied would surely be annoying. 

Closing for now, but feel free to let me know if I've misunderstood the intent behind this one or disagree with my reasoning : )
